CIP-0054/README.md

```diff
@@ -0,0 +1,267 @@
+---
+CIP: 54
+Title: Plutus Contract Blueprint
+Authors: KtorZ <matthias.benkort@cardanofoundation.org>, scarmuega <santiago@carmuega.me>, satran004
+Discussions-To: https://github.com/cardano-foundation/CIPs/pull/258
+Status: Draft 
+Type: Process
+Created: 2022-05-15
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This document specifies a language for documenting Plutus contracts in a machine-readable manner. This is akin to what [OpenAPI](https://swagger.io/specification) or [AsyncAPI](https://www.asyncapi.com/docs/specifications/v2.4.0) are for, documenting HTTP services and asynchronous services respectively. In a similar fashion, Plutus contracts have an interface which is mostly defined by its datums and redeemers structures. Plus, contracts evolving around state-machines can be represented as a serie of state transitions. This document is therefore a meta-specification fixing the vocabulary by which one can specify a Plutus contract interface.
+
+## Motivation
+
+While publicly accessible, on-chain contracts are currently inscrutable. Ideally, one would want to get an understanding of transactions surrounding certain script executions. This is both useful to visualize and control the evolution of a contract life-cycle; but also, as a user interacting with a contract, to ensure that I am authorizing a transaction to do what I intend to. Having a machine-readable specification in the form of a JSON-schema makes it easier (possible) to create a wide variety of use-cases around a single concise document, such as:
```

Comment:
This seems tricky, given that many applications have wildly differing structures and interaction models. I think you may need to make some fairly big assumptions about how the interaction models are structured in order to get a usable language for describing the different ways you can interact with things.

Reply:
So far, I think we're mainly focusing on two things, the first being relatively ubiquitous:

1. Get a machine-readable description of the datum and redeemers, which is basically just a JSON-schema for the Plutus `Data` (see above).

2. For state-machine based contracts, a way to represent transitions between states of the state-machine. Surely, this only applies to a subset of the applications but perhaps long term, may also encourage more and more DApps to organize around state-machines if it gives them access to a variety of tools built on top :)

Reply:
Note that the first point in itself, while not being anything revolutionary, has already value when it comes to automation and building capable interfaces.